CONN-2959 Beta Config Based C2C (#3800)
SumoLogic/sumologic-documentation: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/config-based-source.md

---
id: config-based-source
title: Config Based Source
sidebar_label: STIX/TAXII 2 Client
tags:
  - cloud-to-cloud
  - config
  - based
description: Learn how to set up a Config Based C2C to collect data into the Sumo Logic environment.

> ---

Author: I've noticed the title does not appear and I don't know why.

<img width="1597" alt="image" src="https://github.com/SumoLogic/sumologic-documentation/assets/45008325/116dca0a-5786-45ca-9d6f-6de71fe10a25">

Reviewer: It looks like the author used one hash on Setup, which markdown reads as a title, and then itoverrides the actual title. So you just have to add a second hash and that'll fix it.

<img width="350" alt="Screenshot 2024-03-07 at 1 43 09 PM" src="https://github.com/SumoLogic/sumologic-documentation/assets/56411016/0357296c-72fd-48c0-bdd0-13dfb47ee06c">

Reviewer: fixed

import CodeBlock from '@theme/CodeBlock';
import useBaseUrl from '@docusaurus/useBaseUrl';

<head>
  <meta name="robots" content="noindex" />
</head>

<p><a href="/docs/beta"><span className="beta">Beta</span></a></p>

> Sumo Logic provides this configuration based cloud source to collect log data from vendor APIs with a modular configuration. The goal of this source is for Sumo Logic to expand the configuration modules over time giving greater compatibility with vendor APIs, but to acknowledge complex APIs will still require a specific cloud source and not be compatible with this source.

Reviewer: ```suggestion
With our configuration-based (_config-based_) cloud source, you can collect log data from vendor APIs with a modular configuration. The goal of this source is for Sumo Logic to expand the configuration modules over time giving greater compatibility with vendor APIs, but to acknowledge complex APIs will still require a specific cloud source and not be compatible with this source.
```

:::note
This source is available in the [Fed deployment](/docs/api/getting-started#sumo-logic-endpoints-by-deployment-and-firewall-security).
:::

> # Setup

Reviewer: ```suggestion
## Setup
```

Reviewer: fixed

If this source is not able to be configured to support your vendor API, we can either take requests to add additional features to support it or take a request to build a dedicated source integration if needed. 

1. Locate the vendors API docs describing the API.
1. Identify how the API implements authentication.
1. Identify one API endpoint to collect logs from. You will need to create multiple Sumo sources if you wish to collect log data from more than one endpoint.
   - What is the endpoint URL?
   - Are there any required HTTP Headers?
   - What URL parameters are available to use? Which ones track progression such as ingest timestamps?
   - How is pagination implemented? 
   - Where in the response body is the array of logs to ingest?
   - Are there timestamps within the response array of logs indicating the individual log timestamp?

> 1. Identify API rate limits

Reviewer: ```suggestion
1. Identify API rate limits.
```

   - Does the API specify a rate limit?
   - Is bursting allowed?

> ## Source Configuration

Reviewer: ```suggestion
## Source configuration
```

1. (Optional) **Parser path**. If **Forward to SIEM** option is selected, provide a parser path if available.
1. (Optional) **Fields**. Click the **+Add** button to define the fields you want to associate. Each field needs a name (key) and value.
   - ![green check circle.png](/img/reuse/green-check-circle.png) A green circle with a check mark is shown when the field exists in the Fields table schema.
   - ![orange exclamation point.png](/img/reuse/orange-exclamation-point.png) An orange triangle with an exclamation point is shown when the field doesn't exist in the Fields table schema. In this case, an option to automatically add the nonexistent fields to the Fields table schema is provided. If a field is sent to Sumo Logic that does not exist in the Fields schema it is ignored, known as dropped.
1. **Configuration Sections**. Expand each section to learn more about the options available for configuration. 
<details>
  <summary>Authentication</summary>
  <div>
    Choose the type of authentication the source will use based on the vendor API requirements and configure the details of that specific authentication type.

#### Basic

> It is a default value. Select this authentication option if the vendor API requires basic HTTP authentication. The source will always add the `Authorization` HTTP request header with the base64 encoded `username:password` as the value.

Reviewer: ```suggestion
Basic is the default value. Select this authentication option if the vendor API requires basic HTTP authentication. The source will always add the `Authorization` HTTP request header with the base64 encoded `username:password` as the value.
```

> - **Basic HTTP Username**: The username required by the vendor API.
> - **Basic HTTP Password**: The password required by the vendor API.

Reviewer: ```suggestion
- **Basic HTTP Username**. The username required by the vendor API.
- **Basic HTTP Password**. The password required by the vendor API.
```

---
id: config-based-source
title: Config Based Source

> sidebar_label: STIX/TAXII 2 Client

Reviewer 2: ```suggestion
sidebar_label: Config Based Source
```
